gaphor/diagram/general/diagramitem.py

```diff
@@ -0,0 +1,49 @@
+"""
+An item representing a diagram.
+"""
+
+from gaphor.core.modeling.diagram import Diagram
+from gaphor.diagram.presentation import ElementPresentation, Named
+from gaphor.diagram.shapes import IconBox, Text, Box, stroke, draw_border
+from gaphor.diagram.support import represents
+from gaphor.diagram.text import FontWeight
+
+
+@represents(Diagram)
+class DiagramItem(ElementPresentation, Named):
+    def __init__(self, diagram, id=None):
+        super().__init__(diagram, id, width=30, height=30)
+        for h in self.handles():
+            h.movable = False
+
+        self.shape = IconBox(
+            Box(
+                Box(draw=draw_diagram),
+                style={
+                    "padding": (4, 4, 4, 4),
+                    "border-radius": 4,
+                },
+                draw=draw_border,
+            ),
+            Text(
+                text=lambda: self.subject and self.subject.diagramType or "",
```

Comment:
I really like showing the diagram abbreviation below the icon, this looks great!